sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala

```diff
@@ -0,0 +1,709 @@
+      .map(view.updateAccountStorageBytes(contractAddress, nodeId, _))
+  }
+
+  def removeForgerStake(view: BaseAccountStateView, stakeId: Array[Byte]): Unit = {
+    val nodeToRemoveId = Blake2b256.hash(stakeId)
+
+    // we assume that the caller have checked that the forger stake really exists in the stateDb.
+    // in this case we must necessarily have a linked list node
+    val nodeToRemove = findLinkedListNode(view, nodeToRemoveId).get
+
+    // modify previous node if any
+    modifyNode(view, nodeToRemove.previousNodeKey) { previousNode =>
+      LinkedListNode(previousNode.dataKey, previousNode.previousNodeKey, nodeToRemove.nextNodeKey)
+    }
+
+    // modify next node if any
+    modifyNode(view, nodeToRemove.nextNodeKey) { nextNode =>
+      LinkedListNode(nextNode.dataKey, nodeToRemove.previousNodeKey, nextNode.nextNodeKey)
+    } getOrElse {
+      // if there is no next node, we update the linked list tip to point to the previous node, promoted to be the new tip
+      view.updateAccountStorage(contractAddress, LinkedListTipKey, nodeToRemove.previousNodeKey)
+    }
+
+    // remove the stake
+    view.removeAccountStorageBytes(contractAddress, stakeId)
+
+    // remove the node from the linked list
+    view.removeAccountStorageBytes(contractAddress, nodeToRemoveId)
+  }
+
+  def getListItem(view: BaseAccountStateView, tip: Array[Byte]): (AccountForgingStakeInfo, Array[Byte]) = {
+    if (!linkedListNodeRefIsNull(tip)) {
+      val node = findLinkedListNode(view, tip).get
+      val stakeData = findStakeData(view, node.dataKey).get
+      val listItem = AccountForgingStakeInfo(
+        node.dataKey,
+        ForgerStakeData(
+          ForgerPublicKeys(
+            stakeData.forgerPublicKeys.blockSignPublicKey, stakeData.forgerPublicKeys.vrfPublicKey),
+          stakeData.ownerPublicKey, stakeData.stakedAmount)
+      )
+      val prevNodeKey = node.previousNodeKey
+      (listItem, prevNodeKey)
+    } else {
+      throw new ExecutionRevertedException("Tip has the null value, no list here")
+    }
+  }
+
+  def linkedListNodeRefIsNull(ref: Array[Byte]): Boolean =
+    BytesUtils.toHexString(ref).equals(BytesUtils.toHexString(LinkedListNullValue))
+
+  override def addScCreationForgerStake(msg: Message, view: BaseAccountStateView): Array[Byte] =
+    doAddNewStakeCmd(msg, view, isGenesisScCreation = true)
+
+  def doAddNewStakeCmd(msg: Message, view: BaseAccountStateView, isGenesisScCreation: Boolean = false): Array[Byte] = {
+
+    // check that message contains a nonce, in the context of RPC calls the nonce might be missing
+    if (msg.getNonce == null) {
+      throw new ExecutionRevertedException("Call must include a nonce")
+    }
+
+    // check that msg.value is greater than zero
+    if (msg.getValue.signum() <= 0) {
+      throw new ExecutionRevertedException("Value must not be zero")
+    }
+
+    // check that msg.value is a legal wei amount convertible to satoshis without any remainder
```

Comment:
Good point, I thought this is because the staking somehow requires the amounts to be representable in zennies, but I don't see a conversion happening here.

Reply:
right, stakes must be representable in zennies. So we need this check.
Store in wei to make it eth frienldy, when someone will take a look at forgers stakes using standard explorer.